src/JsonService.ts

```diff
@@ -177,4 +186,29 @@ export class JsonService {
 
         return json;
     }
+
+    private appendExtraHeaders(
+        headers: Record<string, string>,
+    ): void {
+        const customKeys = Object.keys(this._extraHeaders);
+        const protectedHeaders = [
+            "authorization",
+            "accept",
+            "content-type",
+        ];
+        if (customKeys.length === 0) {
+            return;
+        }
+        customKeys.forEach((headerName) => {
+            if (protectedHeaders.includes(headerName.toLocaleLowerCase())) {
+                return;
```

Comment:
at least log an warning here

Reply:
ok

Reply:
`logger.warn("Protected header could not be overridden", headerName, protectedHeaders);`
 
It's good for you ?